### DISCUSSION TITLE
kata 4 bucle for

### DISCUSSION THREAD

**Discussion by larap03** [LaunchX-InnovaccionVirtual/CursoIntroPython#306] [Q&A] [answered]
por qué salen tantas n?
![image](https://user-images.githubusercontent.com/99061466/154622978-1c7bb2e2-4b55-4347-bd63-57624a636c89.png)

**Comment by doguedogue** [comment]
Tienes mal el segundo ciclo
For key_word in key_word

Debería ser
For key_word in sentence

**Comment by JavierMendez-Coder** [comment, marked as answer]
Buenas noches. 
Tu problema surge de que tu arreglo de palabras clave es una sola cadena en vez de varias palabras. Deja me explico:
```python
# Así lo tienes:
key_word = ["average, temperature, distance"]

# Así debería de ser:
key_word = ["average", "temperature", "distance"] # <-- Observa que cada palabra se encuentra entre comillas (").
```
Además, estas repitiendo el nombre de la variable con la que recorres el arreglo:
```python
key_word = ["average", "temperature", "distance"]

for sentence in text:
  for key_word in key_word: # <-- Aquí 
    if key_word in sentence:
      print(sentence)
      break
```
Y debería ser algo así:
```python
key_words = ["average", "temperature", "distance"] # <-- Observa que cambié ligeramente el nombre del arreglo.

for sentence in text:
  for key_word in key_words: # <-- Observa que el nombre de las variables no es el mismo.
    if key_word in sentence:
      print(sentence)
      break
```
Y por último, no divides el arreglo en múltiples secciones, es decir, no utilizas `split().`
```python
text = """Hola. Aquí va el texto en el que quieres utilizar la función split()."""

sentences = text.split(". ")

for sentence in sentences:
  for key_word in key_words:
    if key_word in sentence:
      print(sentence)
      break
```
```terminal
--- Output: ---
Hola
Aquí va el texto en el que quieres utilizar la función split().
```

Espero haberte ayudado. 👋

**Reply by larap03** [reply]
gracias ya entendí el error, ya lo pude solucionar.

**Reply by JavierMendez-Coder** [reply]
En ese caso te agradecería marcar alguna respuesta como solución, para las personas que puedan contar con la misma duda, o en su defecto, escribir como lo solucionaste tu y marcarla.
